src/features/home-calendar/ui/MiniCalendar.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react';
+import { LeftArrow, RightArrow } from 'shared/icons';
+import * as s from './style.css';
+
+const MiniCalendar = () => {
+  const [currentDate, setCurrentDate] = useState(new Date());
+  const [selectedDate, setSelectedDate] = useState<number | null>(
+    new Date().getDate()
+  );
+
+  const startOfMonth = new Date(
+    currentDate.getFullYear(),
+    currentDate.getMonth(),
+    1
+  );
+
+  const endOfMonth = new Date(
+    currentDate.getFullYear(),
+    currentDate.getMonth() + 1,
+    0
+  );
+
+  const startWeek = startOfMonth.getDay();
+  const daysInMonth = endOfMonth.getDate();
+
+  const prevEndOfMonth = new Date(
+    currentDate.getFullYear(),
+    currentDate.getMonth(),
+    0
+  ).getDate();
```

Comment:
날짜들을 JSON 형태로 관리하는 것이 더 효율적일듯 합니다.

```suggestion
   const {
    startOfMonth,
    endOfMonth,
    startWeek,
    daysInMonth,
    prevEndOfMonth
  } = useMemo(() => ({
    startOfMonth: new Date(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      1
    ),
    endOfMonth: new Date(
      currentDate.getFullYear(),
      currentDate.getMonth() + 1,
      0
    ),
    startWeek: new Date(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      1
    ).getDay(),
    daysInMonth: new Date(
      currentDate.getFullYear(),
      currentDate.getMonth() + 1,
      0
    ).getDate(),
    prevEndOfMonth: new Date(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      0
    ).getDate()
  }), [currentDate]);
```

위와 같은 방식으로 날짜 계산이랑 변수들을 합쳐서 메모리 관리에도 용이하고 접근성에서도 방면에서도 편하게 설계하는 것이 더 좋겠습니다.